### ISSUE TITLE
Allen and Heath sq5

### ISSUE THREAD

**Issue by tom-bindon** [bitfocus/companion-module-requests#56]
**Describe the feature**
I would like to be able to control Scene changes, Fader levels and channel mutes on my SQ5 sound desk

**Is this platform dependent (windows, mac, ..)?**
Not sure what you mean 

**If documentation is required to implement, do you know where to find it?**
https://www.allen-heath.com/ahproducts/sq-5/

**Usecases**
It will be useful on a lot of my events where I am quite often operating the SQ5, Qlab and a Chamsys lighting desk together trying to trigger cues at the same time. Using this would allow me to trigger one button for a all of that plus even being able to added the greenhippo in the future would make this even easier on events.

**Comment by josephdadams** [comment]
Protocol: https://www.allen-heath.com/media/SQ-MIDI-Protocol-Issue1.pdf

**Comment by josephdadams** [comment]
@webby123

**Comment by simongroom112** [comment]
Is the mapping for this very similar or the same to the DLive?

**Comment by webby123** [comment]
I Think it's not, it would be nice if we can switch the mutes on/off on the
SQ5

maybe this will help: https://www.allen-heath.com/ahproducts/sq-5/


Best regards,

*Eddy Caron*

Cygnea Congrestechniek
Brasem 46
<https://maps.google.com/?q=Brasem+46+4941+SE%C2%A0+Raamsdonksveer+The+Netherlands&entry=gmail&source=g>
4941 SE  Raamsdonksveer
<https://maps.google.com/?q=Brasem+46+4941+SE%C2%A0+Raamsdonksveer+The+Netherlands&entry=gmail&source=g>
The Netherlands
<https://maps.google.com/?q=Brasem+46+4941+SE%C2%A0+Raamsdonksveer+The+Netherlands&entry=gmail&source=g>

Office   +31 (0)85 02 90 331
Mobile  +31 (0)6 53 69 25 10

e.caron@cygnea-congrestechniek.nl
www.cygnea-congrestechniek.nl  <http://www.cygnea-congrestechniek.nl/>



Wij gaan zorgvuldig met uw gegevens om.
Hoe wij dat doen kunt u hier lezen door onze Privacy Verklaring hier te
downloaden. <https://cygnea-congrestechniek.nl/privacyverklaring.pdf>


Op zo 6 sep. 2020 om 13:54 schreef Ed <webby123@ziggo.nl>:

**Comment by simongroom112** [comment]
Updated Protocol Document:
https://www.allen-heath.com/media/SQ-MIDI-Protocol-Issue2.pdf

**Comment by MeestorX** [comment]
A bounty would definitely make a difference. Time to do free stuff is getting scarce and it's a lot more work when you don't own the device that you're building a module for.
If you are able to put up a bounty then I can see about adding the SQ to the dLive/iLive module or creating a new one, depending on what the admins prefer.
I would also need a commitment to serious testing (including possible zoom conferences if there are bugs) as I do not have an SQ to use for testing. Contact me at www.checkcheckonetwo.com if this sounds acceptable.

**Comment by simongroom112** [comment]
I’ve followed your link and opened a thread but not sure if that was what you wanted me to do.... in the mean time....

Please let me know what sort of bounty you’d consider,

I’d be more than happy to assist with testing and via zoom if required!

**Comment by MeestorX** [comment]
Thank you. Replied in the thread.

**Comment by RoyvanSchelven** [comment]
Hi All, is there any news about this A&H SQ request and the implementation of the module?

**Comment by simongroom112** [comment]
Hi, 

The last few months hadn’t really gone to plan.... I think we last spoke about sponsoring the project... what type of figure would you deem acceptable? 

Kind Regards
Simon Groom

**Comment by simongroom112** [comment]
Someone was happy to look into this but was looking for some form of sponsorship... if anyone wants to put in to a pot?

**Comment by simongroom112** [comment]
Ive started a GoFundMe page to collect sponsorship money for the creation of this module. 

https://www.gofundme.com/f/sq-module-for-companion?utm_source=customer&utm_medium=copy_link&utm_campaign=p_cf+share-flow-1

**Comment by JITZEH** [comment]
Is it ready or takes it longer

**Comment by josephdadams** [comment]
I don’t think anyone has started on it.

**Comment by JITZEH** [comment]
That is why make an New message

**Comment by simongroom112** [comment]
Please consider donating to creation fundraiser above.

**Comment by JITZEH** [comment]
For 300 pounds?! I can make it by myself.

**Comment by simongroom112** [comment]
£300 is an open target. If 10 People put in £10-£20 each it’ll get made quicker.... time is money....

If you can make it yourself why don’t you just make it and submit it? 

Kind Regards
Simon Groom

**Comment by JITZEH** [comment]
Hi guys,
I'm going to try to make it for my internship time. Has anyone already started on this project? Please send me a message.

kind regards,
Jitze van der Hoek

**Comment by fccnorwalktech** [comment]
I have access to an SQ5 and SQ7 for limited testing if someone is able to produce a module.

**Comment by the5spot** [comment]
I use an SQ6. I will contribute to the gofundme page. If I could control scene changes or mute groups with companion.... I'd be thrilled. I am also able to help troubleshoot for the person that works on this

**Comment by the5spot** [comment]
> Hi guys,
> I'm going to try to make it for my internship time. Has anyone already started on this project? Please send me a message.
> 
> kind regards,
> Jitze van der Hoek

Any updates on this yet?

**Comment by josephdadams** [comment]
There was some discussion on the Facebook Companion User Group recently about it. I am looking to be hired to create modules for Companion right now and discussed looking at this protocol, but since the GoFundMe has not reached funding yet, it is not at a point where I can get involved. But perhaps others will be interested in working on it.

**Comment by brynte** [comment]
> There was some discussion on the Facebook Companion User Group recently about it. I am looking to be hired to create modules for Companion right now and discussed looking at this protocol, but since the GoFundMe has not reached funding yet, it is not at a point where I can get involved. But perhaps others will be interested in working on it.

Quite near now, though.

**Comment by the5spot** [comment]
I've tried to donate $50 to the gofundme. It won't let me... Does it allow donations from the US using the dollar?

**Comment by brynte** [comment]
> I've tried to donate $50 to the gofundme. It won't let me... Does it allow donations from the US using the dollar?

Gofundme is an american company. It shouldn't be a problem. However A&H is british. I guess that's why the fundraiser is in pounds.

**Comment by simongroom112** [comment]
@MeestorX can you email me at simon.groom@me.com please? I can't log onto your site to contact you! Cheers

**Comment by JITZEH** [comment]
> > Hi guys,
> > I'm going to try to make it for my internship time. Has anyone already started on this project? Please send me a message.
> > kind regards,
> > Jitze van der Hoek
> 
> Any updates on this yet?

no. sorry guys companion dev works not great for MACOS. I don't have windows computer nearby to make it soon. 
sorry...

**Comment by JITZEH** [comment]
update protocol: https://www.allen-heath.com/media/SQ-MIDI-Protocol-Issue3.pdf

**Comment by JITZEH** [comment]
with this module works scene recalling.
https://github.com/bitfocus/companion-module-allenheath-dlive

**Comment by MeestorX** [comment]
Looks like the bounty is getting up there. I would be willing to work on this, but I've never done a gofundme. How does one ensure that the person doing the work gets the money when the work is done?
Oh, I'd also need at least one person who has the console continuously available to them (i.e. in their home) that would commit to testing builds for me.

**Comment by simongroom112** [comment]
I am running the GoFundMe page and so the raised funds are transferred to my bank account. So I could then transfer this to you with/without invoice?

I own and can have constant access to an SQ5 so am willing to sit on zoom with you :-) 

Kind Regards
Simon Groom

**Comment by MeestorX** [comment]
Ok, sounds like a plan. I'll continue the conversation via e-mail with you.

**Comment by fccnorwalktech** [comment]
Since this is almost there, I will finish it off so that development can start.

Again I have access to an sq5 and sq7 to help develop as needed.

**Comment by JITZEH** [comment]
@MeestorX nice to see you That want to do it. (I have also a SQ5)

**Comment by the5spot** [comment]
I have the SQ6 and contributed to the GoFundMe.... I'd love to help out with the testing. my email is the5spotnashville@gmail.com . Thanks!

**Comment by the5spot** [comment]
What are the possible controls with this? Everything I imagine controlling could be done with scene changes and mute groups... but maybe there are more possibilities.

**Comment by JITZEH** [comment]
> What are the possible controls with this? Everything I imagine controlling could be done with scene changes and mute groups... but maybe there are more possibilities. 

Same things as this module: https://github.com/bitfocus/companion-module-allenheath-dlive

**Comment by MeestorX** [comment]
Thanks, everyone. The SQ is a completely different beast communication and feature-wise, but the basic operation within Companion should be similar to the dLive.

**Comment by JITZEH** [comment]
> Thanks, everyone. The SQ is a completely different beast communication and feature-wise, but the basic operation within Companion should be similar to the dLive.

Yea. (a tip from me the. scene recall works already if you put the data from the SQ mixer in the ilive/Dlive section.)

**Comment by MeestorX** [comment]
Looks like I was beaten to it! :-)
https://github.com/bitfocus/companion-module-allenheath-sq

**Comment by JITZEH** [comment]
> Looks like I was beaten to it! :-)
> 
> https://github.com/bitfocus/companion-module-allenheath-sq
> 
> 

Is it already tested?

**Comment by MeestorX** [comment]
> Is it already tested?

I would imagine he has one of those consoles and has tested it, but you'll have to ask him. Post a new "issue" on his repository to ask about it.

**Comment by JITZEH** [comment]
> > Is it already tested?
> 
> 
> 
> I would imagine he has one of those consoles and has tested it, but you'll have to ask him. Post a new "issue" on his repository to ask about it.
> 
> 
> 
> 
Thank you!

**Comment by maxks** [comment]
> > Looks like I was beaten to it! :-)
> > https://github.com/bitfocus/companion-module-allenheath-sq
> 
> Is it already tested?

I developed this module initially for my personal use, so, yes it tested on SQ5 during development.

**Comment by fccnorwalktech** [comment]
@maxks Any idea when this may get added to the beta builds? or how do we add this to companion currently?

**Comment by maxks** [comment]
> @maxks Any idea when this may get added to the beta builds? or how do we add this to companion currently?

No, I don't know when it release. At this moment, the only way to use this module on Companion is in develop mode.

**Comment by MeestorX** [comment]
> > @maxks Any idea when this may get added to the beta builds? or how do we add this to companion currently?
> 
> No, I don't know when it release. At this moment, the only way to use this module on Companion is in develop mode.

Ask them to include it into the latest build on the #module-development channel in Slack.

**Comment by maxks** [comment]
Jeffrey has response yesterday to a question in #companion and said "it's on the list".

**Comment by maxks** [comment]
Module is now in Companion Beta release.

**Comment by the5spot** [comment]
Sorry for the noob question.... How do I add this onto my rasp pi ??

**Comment by JITZEH** [comment]
> Sorry for the noob question.... How do I add this onto my rasp pi ??

Which OS do you use?

**Comment by JITZEH** [comment]
> Module is now in Companion Beta release.

nice !!!

**Comment by the5spot** [comment]
Raspberry Pi OS.   I installed it from this https://github.com/bitfocus/companion/wiki/CompanionPi-Documentation

**Comment by JITZEH** [comment]
> Raspberry Pi OS. I installed it from this https://github.com/bitfocus/companion/wiki/CompanionPi-Documentation

but this didn't work?

**Comment by the5spot** [comment]
It works great, but it does not have the Allen and Heath on it, neither does the "companion-update",  that is in the beta builds.

**Comment by JeffreyDavidsz** [comment]
The update script in Companion PI is fixed in updates because you need to update node first;

Do a `n 12.19.1`
Then `git pull`
Then `yarn update`
The commands needs to be done in the Companion's directory

**Comment by JeffreyDavidsz** [comment]
I see I use 12.18.4 myself

**Comment by the5spot** [comment]
I have it working on my Pi, Mac, and PC... BUT only can connect 1 at a time. 
when I connect another, this error comes up

25. 14:34:50 instance(SQ): MIDI error: read ECONNRESET

**Comment by maxks** [comment]
I think it's a limit of the console that not support multiple MIDI connections at the same time.

**Comment by zcarnefix** [comment]
Is it possible to specify the MIDI channel of the SQ within Companion to be able to use more than 1 at a time? I have several SQ's that I would love to control simultaneously if possible.

**Comment by maxks** [comment]
@zcarnefix new beta released today with that option in module configuration.

**Comment by Djiaant** [comment]
How do we implement the beta module into Companion?

I am trying to control an A&H SQ5 and see that a module was made yet is not included on the companion app to use with a Stream Deck.  Or is this not a thing?

Newbie here!

**Comment by josephdadams** [comment]
I believe it’s included. At least in the beta builds and looks like it has been for awhile. What version of Companion do you use?

**Comment by josephdadams** [comment]
Transferring to the sq repo.
